OpenDreamRuntime/DreamValue.cs

```diff
@@ -108,7 +108,7 @@ public enum DreamValueTypeFlag {
 
         public bool IsNull {
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get => Type == DreamValueType.DreamObject && _refValue == null;
+            get => Type == DreamValueType.DreamObject && (_refValue == null || ((DreamObject) _refValue).Deleted);
```

Comment:
Since we're already checking Type you can probably use `Unsafe.As` to avoid a typecast on every `IsNull` call